### PR TITLE
GH-893: Reactivate ECMA test suite

### PR DESCRIPTION
**Pull request by mmews-n4**
see #893

### CODE REVIEW

File: testhelpers/org.eclipse.n4js.tests.helper/src/org/eclipse/n4js/N4JSStandaloneTestsModule.java [hunk 1]
```diff
@@ -28,6 +29,23 @@
  */
 @XpectGuiceModule
 public class N4JSStandaloneTestsModule extends BaseTestModule {
+
+	/** Constructor enables JS support */
+	public N4JSStandaloneTestsModule() {
+		N4JSLanguageUtils.OPAQUE_MODULE_SUPPORTED = false;
+
+		// Following does not work
```
Comment (yoosiba) [inline testhelpers/org.eclipse.n4js.tests.helper/src/org/eclipse/n4js/N4JSStandaloneTestsModule.java]: delete

**Comment by mmews-n4** [comment]
@yoosiba Would have a second look?

File: testhelpers/org.eclipse.n4js.tests.helper/src/org/eclipse/n4js/N4JSStandaloneTestsModule.java [hunk 2]
```diff
@@ -28,6 +28,12 @@
  */
 @XpectGuiceModule
 public class N4JSStandaloneTestsModule extends BaseTestModule {
+
+	/** Constructor enables JS support */
+	public N4JSStandaloneTestsModule() {
```
Comment (yoosiba) [inline testhelpers/org.eclipse.n4js.tests.helper/src/org/eclipse/n4js/N4JSStandaloneTestsModule.java]: this is causing JS mode being enabled in lots of places. Also transitively via `N4JSInjectorProvider` and others.
In essence we are testing product with JS support and deliver product with not JS  support.
Could we narrow the scope enabling JS mode?

Comment (yoosiba) [reply testhelpers/org.eclipse.n4js.tests.helper/src/org/eclipse/n4js/N4JSStandaloneTestsModule.java]: If needed, have it in separate packages or even bundle?

File: tests/org.eclipse.n4js.transpiler.es.tests/src/org/eclipse/n4js/transpiler/es/tests/AbstractTranspilerTest.xtend [hunk 3]
```diff
@@ -56,19 +57,19 @@ abstract class AbstractTranspilerTest {
...
-
 	@Inject private EcmaScriptTranspiler esTranspiler;
-
+	@Inject	private TranspilerDebugUtils transpilerDebugUtils;
 	@Inject protected EcmaScriptSubGenerator esSubGen
 
-	@Inject
-	private TranspilerDebugUtils transpilerDebugUtils;
+	new() {
+		JSActivationUtil.enableJSSupport();
```
Comment (yoosiba) [inline tests/org.eclipse.n4js.transpiler.es.tests/src/org/eclipse/n4js/transpiler/es/tests/AbstractTranspilerTest.xtend]: again, for me, too wide scope

File: testhelpers/org.eclipse.n4js.tests.helper/src/org/eclipse/n4js/N4JSStandaloneTestsModule.java (hunk 2 above)
Comment (mmews-n4) [reply testhelpers/org.eclipse.n4js.tests.helper/src/org/eclipse/n4js/N4JSStandaloneTestsModule.java]: > In essence we are testing product with JS support and deliver product with not JS support.

-> Yes, that's exactly what we want.

**Pull request merged**